File: go/agent.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// AgentService — V3 agent security.
type AgentService struct {
	t *internal.Transport
}

type AgentScanRequest struct {
	Prompt       string         `json:"prompt"`
	ExternalData []string       `json:"external_data"`
	SessionID    string         `json:"session_id"`
	UserID       string         `json:"user_id"`
	AgentType    string         `json:"agent_type,omitempty"`
	Context      map[string]any `json:"context,omitempty"`
}

func (s *AgentService) Scan(ctx context.Context, req AgentScanRequest) (map[string]any, error) {
	if req.ExternalData == nil {
		req.ExternalData = []string{}
	}
	if req.SessionID == "" {
		req.SessionID = "smoke-session"
	}
	if req.UserID == "" {
		req.UserID = "sdk-smoke"
	}
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/agent/scan", req, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *AgentService) Toolchain(ctx context.Context, tools []map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/agent/toolchain", map[string]any{"tools": tools}, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *AgentService) MemoryPoisoning(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/agent/memory-poisoning", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *AgentService) ReasoningHijack(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/agent/reasoning-hijack", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *AgentService) ToolDisguise(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/agent/tool-disguise", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

File: python/src/aegis/resources/agent.py
"""Agent resource — V3 agent security endpoints.

Server expects ``prompt`` (or ``agent_input``), ``external_data`` (array of
strings, also accepted as ``tools``), ``session_id``, ``user_id``, optional
``agent_type`` and ``context``.
"""

from __future__ import annotations

import uuid
from typing import Any, Dict, List, Optional

from aegis.resources._base import AsyncResource, SyncResource


def _scan_body(
    prompt: str,
    *,
    external_data: Optional[List[str]] = None,
    session_id: Optional[str] = None,
    user_id: Optional[str] = None,
    agent_type: Optional[str] = None,
    context: Optional[Dict[str, Any]] = None,
    **kwargs: Any,
) -> Dict[str, Any]:
    body: Dict[str, Any] = {
        "prompt": prompt,
        "external_data": external_data or [],
        "session_id": session_id or str(uuid.uuid4()),
        "user_id": user_id or "sdk-smoke",
    }
    if agent_type is not None:
        body["agent_type"] = agent_type
    if context is not None:
        body["context"] = context
    body.update(kwargs)
    return body


class SyncAgent(SyncResource):

    def scan(self, prompt: str, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/agent/scan", json=_scan_body(prompt, **kwargs))

    def toolchain(self, tools: List[Dict[str, Any]], **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/agent/toolchain", json={"tools": tools, **kwargs})

    def memory_poisoning(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/agent/memory-poisoning", json=kwargs)

    def reasoning_hijack(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/agent/reasoning-hijack", json=kwargs)

    def tool_disguise(self, **kwargs: Any) -> Dict[str, Any]:
        return self._post("/v3/agent/tool-disguise", json=kwargs)


class AsyncAgent(AsyncResource):

    async def scan(self, prompt: str, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/agent/scan", json=_scan_body(prompt, **kwargs))

    async def toolchain(self, tools: List[Dict[str, Any]], **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/agent/toolchain", json={"tools": tools, **kwargs})

    async def memory_poisoning(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/agent/memory-poisoning", json=kwargs)

    async def reasoning_hijack(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/agent/reasoning-hijack", json=kwargs)

    async def tool_disguise(self, **kwargs: Any) -> Dict[str, Any]:
        return await self._post("/v3/agent/tool-disguise", json=kwargs)
